src/Printer8.php

```diff
@@ -2,9 +2,48 @@
 
 namespace mheap\GithubActionsReporter;
 
+use PHPUnit\Framework\TestFailure;
+use PHPUnit\Framework\TestResult;
 use PHPUnit\TextUI\ResultPrinter;
 
+use function mheap\GithubActionsReporter\Functions\getCurrentType;
+use function mheap\GithubActionsReporter\Functions\printDefects;
+use function mheap\GithubActionsReporter\Functions\printDefectTrace;
+
 class Printer8 extends ResultPrinter
 {
-    use Trait8;
```

Comment:
What was the thinking behind removing the Trait here? `Printer8` and `Printer9` look like duplications to me now

Reply:
`Printer9` and `Printer8` extend different classes, is the only difference - otherwise `Printer9` could just extend `Printer8`.
I left them both as individual concrete classes otherwise just in case adjustments need to be made specific to that version in the future.
(please note I am personally biased against the use of traits for various reasons).
I _could_ make an abstract base class, not extending PHPUnit to avoid this duplication, if preferred?

Reply:
Using a trait allows us to keep the functionality DRY whilst still extending from different classes.

What are the benefits/risks of using a base class rather than a trait in this case?

Reply:
Because the method signatures vary over the PHPUnit versions, and traits are not actually concrete testable structures, sort of invisible magic.

Trait8 & Printer7 are already repeating logic because of this.
It would be a valid use case of traits if the method signatures were the same, but you'll end up with less codeby decorating the common methods with helper functions or something alike